docs/reference/index-modules/indexing-pressure.asciidoc

```diff
@@ -0,0 +1,56 @@
+[[index-modules-indexing-pressure]]
+== Indexing Pressure
+
+Indexing documents into Elasticsearch introduces system load in the form of
+memory and CPU load. Each indexing operation includes coordinating, primary, and
+replica stages. These stages can be performed across multiple nodes in the
+cluster. If too much indexing work is introduced into the system, the cluster
+can become saturated.
+
+Indexing pressure is primarily generated by external operations such as indexing
+requests or CCR follow tasks. However, there is some internal indexing pressure
+during a shard recovery or primary failover.
+
+Elasticsearch internally monitors indexing load. When the load exceeds
+certain limits, new indexing work will be rejected.
+
+[float]
+=== Indexing Stages
+
+External indexing operations go through three stages. The node receiving the
+indexing request is the coordinating node. In the coordinating stage the node
+performs any configured ingest pipelines, separates the request into individual
+shard requests, and dispatches the requests to the primary shards.
+
+In the primary stage, the primary shard node ensures that it is still the primary
+(otherwise it reroutes the request to the new primary), waits on active shards,
+indexes the documents into its <<index-modules-translog,Translog>> and Lucene, and
+dispatches to the replicas.
+
+Finally, in the replica stage, every available replica node indexes the documents
+into its <<index-modules-translog,Translog>> and Lucene.
+
+
+[float]
+=== Memory Limits
+
+Elasticsearch exposes a node setting `indexing_pressure.memory.limit` which
+restricts the number of bytes for outstanding indexing requests. Be default,
+this setting is configured to be 10% of the heap.
```

Comment:
I wonder if we should add here that this setting is already generously sized, and that increasing it too much will have adverse effects on the rest of the system?